internal/failure_converter.go

```diff
@@ -286,7 +295,8 @@ func (dfc *DefaultFailureConverter) FailureToError(failure *failurepb.Failure) e
 			Endpoint:         info.GetEndpoint(),
 			Service:          info.GetService(),
 			Operation:        info.GetOperation(),
-			OperationID:      info.GetOperationId(),
+			OperationToken:   token,
+			OperationID:      token,
 		}
 	} else if info := failure.GetNexusHandlerFailureInfo(); info != nil {
```

Comment:
Don't we need to handle the retry policy here?

Reply:
Yes, thanks!

Reply:
I forgot to add this to the api.

Reply:
I understand we can't have a full integration test, but can we at least have a test that the retry-able flag is being properly serialized and deserialized?

Reply:
Yeah for this PR could we just have a unit test for the failure converter, we should have that anyway regardless of our E2E test coverage

Reply:
Coming in the next commit.